### PR TITLE
Detect library name clashes

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by VoR0220** [comment]
Can you rerun this @chriseth ? I think Jenkins didn't like it....either that or you're doing something that's breaking Jenkins and would like to understand what it is you're breaking.

File: libsolidity/interface/CompilerStack.cpp [hunk 1]
```diff
@@ -400,6 +403,36 @@ void CompilerStack::resolveImports()
...
+							errinfo_secondarySourceLocation(SecondarySourceLocation().append(
+									"The other declaration is here:", libraries[contract->name()]
+							));
+
+						m_errors.push_back(err);
+					}
+					else
+						libraries[contract->name()] = contract->location();
+				}
+	return !clashFound;
```
Comment (yann300) [inline libsolidity/interface/CompilerStack.cpp:434]: seems to always return true

Comment (chriseth) [inline libsolidity/interface/CompilerStack.cpp:434]: Fixed.

**Comment by VoR0220** [comment]
please rebase

**Comment by chriseth** [comment]
Rebased.

**Comment by VoR0220** [comment]
When taken into account with the other PRs (specifically the import methodology), this works excellently. Just make sure that this builds on Ubuntu and you'll be all set to go.

**Pull request merged**